recipes/python/python.inc

```diff
@@ -1,49 +1,266 @@
-# -*- mode:python; -*-
-DESCRIPTION = "Python is a programming language that lets you work more quickly and integrate your systems more effectively."
+DESCRIPTION = "Python is a programming language that lets you work quickly and integrate systems more effectively."
 HOMEPAGE = "https://www.python.org/"
-LICENSE = "python"
-
+LICENSE = "python-2.0"
 RECIPE_TYPES = "native machine"
+COMPATIBLE_HOST_ARCHS = ".*linux"
 
-inherit autotools  make
+# Make sure to set these in the version file
+MAJOR_PV = ""
+MINOR_PV = ""
+MINOR_M  = ""
 
-CROSS_DEPENDS = "native:python  native:python-lib libssl libz libreadline libffi libcrypto libtermcap libm"
-CROSS_DEPENDS:native = ""
-DEPENDS += "${CROSS_DEPENDS}"
+# It's important to inherit auto-package-libs before
+# autotools to make sure the libs end up in the right
+# packages
+inherit auto-package-utils auto-package-libs chrpath autotools
 
-DEPENDS_${PN}:machine += " libz libssl libreadline libffi libcrypto libtermcap libm "
-RDEPENDS_${PN}:machine += "libz libssl libreadline libffi libcrypto libtermcap libm "
+# The native build is only used when building for machine
+# target, so keep dependencies to a minimum
+DEPENDS += "libffi"
 
-SRC_URI = "http://www.python.org/ftp/python/${PV}/Python-${PV}.tar.xz"
-# The machine needs to be patched
-# Original patch from:
-# http://www.trevorbowen.com/2013/10/07/cross-compiling-python-2-7-5-for-embedded-linux/
-# http://bugs.python.org/file31991/Python-2.7.5-xcompile.patch
-# Note the patch has been modified!
-SRC_URI:>machine = " file://Python-${PV}-xcompile.patch;striplevel=3"
+# When building for machine target, include the full
+# dependency list
+DEPENDS:>machine = " \
+    native:python${MAJOR_PV} \
+    native:python${MAJOR_PV}-dev \
+    native:python${MAJOR_PV}-distutils \
+    native:python${MAJOR_PV}-pgen \
+    native:python${MAJOR_PV}-freeze-importlib \
+    libffi \
```

Comment:
Duplicate libffi dependency here.  Either change to DEPENDS:machine or remove this line.